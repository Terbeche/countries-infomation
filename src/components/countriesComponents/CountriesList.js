import { useSelector } from 'react-redux';
import CountriesDisplay from './CountriesDisplay';

const CountriesList = () => {
  const { Countries } = useSelector((state) => state.countries);

  return (

    <div>
      {Countries.map((Country) => (
        <CountriesDisplay
          key={Country.id}
          id={Country.id}
          name={Country.name}
          population={Country.population}
          area={Country.area}
          capital={Country.capital}
          continent={Country.continent}
          flag={Country.flag}

        />
      ))}
    </div>

  );
};

export default CountriesList;
