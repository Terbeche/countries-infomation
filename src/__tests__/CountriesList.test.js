import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/configureStore';
import CountriesList from '../components/countriesComponents/CountriesList';

it('CountriesDisplay component snapshot testing', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <CountriesList />
        </BrowserRouter>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
