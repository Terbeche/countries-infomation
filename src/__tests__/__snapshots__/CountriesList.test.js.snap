// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountriesDisplay component snapshot testing 1`] = `
<div
  className="countries-container"
/>
`;
